src/test/datascience/debugger.vscode.test.ts

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+'use strict';
+import * as sinon from 'sinon';
+import { DebugProtocol } from 'vscode-debugprotocol';
+import { ICommandManager, IVSCodeNotebook } from '../../client/common/application/types';
+import { IDisposable } from '../../client/common/types';
+import { Commands } from '../../client/datascience/constants';
+import { IVariableViewProvider } from '../../client/datascience/variablesView/types';
+import { IExtensionTestApi } from '../common';
+import { initialize, IS_REMOTE_NATIVE_TEST, IS_WEBVIEW_BUILD_SKIPPED } from '../initialize';
+import {
+    canRunNotebookTests,
+    closeNotebooks,
+    closeNotebooksAndCleanUpAfterTests,
+    createEmptyPythonNotebook,
+    insertCodeCell,
+    prewarmNotebooks,
+    workAroundVSCodeNotebookStartPages,
+    waitForEvent,
+    getCellOutputs
+} from './notebook/helper';
+import { verifyViewVariables } from './variableView/variableViewHelpers';
+import { ITestVariableViewProvider } from './variableView/variableViewTestInterfaces';
+import { traceInfo } from '../../client/common/logger';
+import { sleep } from '../core';
+import { IDebuggingManager } from '../../client/debugger/types';
+import { assert } from 'chai';
+import { DebugSession } from 'vscode';
+import { KernelDebugAdapter } from '../../client/debugger/jupyter/kernelDebugAdapter';
+import { ITestWebviewHost } from './testInterfaces';
+
+suite('DataScience - Debugging', function () {
+    let api: IExtensionTestApi;
+    const disposables: IDisposable[] = [];
+    let commandManager: ICommandManager;
+    let variableViewProvider: ITestVariableViewProvider;
+    let vscodeNotebook: IVSCodeNotebook;
+    let debuggingManager: IDebuggingManager;
+    this.timeout(120_000);
+    suiteSetup(async function () {
+        traceInfo(`Start Test Suite`);
+        this.timeout(120_000);
+        api = await initialize();
+
+        // We need to have webviews built to run this, so skip if we don't have them
+        if (IS_WEBVIEW_BUILD_SKIPPED) {
+            console.log('Debugging tests require webview build to be enabled (for the variable view)');
+            return this.skip();
+        }
+
+        // Don't run if we can't use the native notebook interface
+        if (IS_REMOTE_NATIVE_TEST || !(await canRunNotebookTests())) {
+            return this.skip();
+        }
+        await workAroundVSCodeNotebookStartPages();
+        await closeNotebooksAndCleanUpAfterTests(disposables);
+        await sleep(5_000);
```

Comment:
Why a sleep?

Reply:
to give time for setup, I'm not 100% sure why 5 seconds and not 1 for example. That is on all out tests.

Reply:
I could only find one test that does this, not everywhere, I think we should remove this.